rig-core/src/embeddings/tool.rs

```diff
@@ -0,0 +1,25 @@
+use crate::{self as rig, tool::ToolEmbeddingDyn};
```

Comment:
The only way I  found to add tools to the embeddings builder was by converting all tools in a `Toolset` to the type `EmbeddableTool` which implements `Embeddable`.

Tool items defined in `Toolset` can all be of different types due to dynamic dispatch. However, I haven't implemented dynamic dispatch for items in the embeddings builder. Therefore, all documents added to the embeddings builder need to be of the same type (that implements `Embeddable`). So before added tools from `Toolset` to `EmbeddingsBuilder`, they are all converted to `EmbeddableTool`. @cvauclair what do you think?